### ISSUE TITLE
feat(v2): GPU support — float64/CUDA setup, array placement, packaging

### ISSUE THREAD

**Issue by darothen** [darothen/pyrcel#41]
## Context

JAX makes GPU support mostly transparent, but two specific issues require attention: float64 on CUDA requires explicit configuration, and the codebase has mixed `np.asarray` / `jnp.asarray` calls that cause silent host-device copies. The `r³ - r_dry³` cancellation fix (issue #35) also improves float32 accuracy, which is relevant here.

### Code changes

- [ ] Audit all uses of `np.asarray` in the JAX codepath (`thermo.py`, `integrator.py`, `equilibrate.py`, `model.py`) and replace with `jnp.asarray` to avoid silent host-device copies
- [ ] Add a device-selection utility (or document the pattern) for explicit CPU vs. GPU placement:
  ```python
  # Recommended: run on GPU if available, fall back to CPU
  jax.default_device(jax.devices('gpu')[0] if jax.devices('gpu') else jax.devices('cpu')[0])
  ```
- [ ] Document float64 on CUDA: `jax_enable_x64=True` is set at module import, but CUDA requires additional setup; add a note in the module docstring of `pyrcel/thermo.py` and in the docs

### Packaging

- [ ] Add `gpu` extra to `pyproject.toml`: `jax[cuda12]`
- [ ] Add a GPU CI job (optional, runs on self-hosted GPU runner if available; skipped otherwise)
- [ ] Add a `examples/gpu_run.py` demo script showing device selection and performance comparison

### Documentation

- [ ] Add a 'GPU Setup' section to the docs covering: CUDA version requirements, float64 caveat, expected speedup profile

## Notes

- float64 is **required** for numerical accuracy (supersaturation evolves on ~1e-5 scales; water vapor mixing ratio changes by ~1e-8 relative). The model cannot run reliably in float32 even with the `Seq` refactor.
- Consumer NVIDIA GPUs have limited float64 throughput; the performance benefit is most pronounced on A100/H100-class hardware.
- Depends on: #33 (module renames), #35 (thermo rewrite)

**Comment by darothen** [comment]
Resolved by 736d404 — `device` parameter on `ParcelModel` and `--device` CLI flag dispatch to GPU. Install with `pip install 'pyrcel[gpu]'`. Merged to master via #82.
